src/viam/sdk/services/mlmodel/client.cpp

```diff
@@ -45,77 +39,36 @@ std::shared_ptr<MLModelService::named_tensor_views> MLModelServiceClient::infer(
     auto* const resp = pb::Arena::CreateMessage<mlpb::InferResponse>(arena.get());
     grpc::ClientContext ctx;
 
-    if (!kUseFlatTensors) {
-        struct tensor_storage_and_views {
-            mlmodel_details::tensor_storage storage;
-            MLModelService::named_tensor_views views;
-        };
-        auto tsav = std::make_shared<tensor_storage_and_views>();
-
-        auto& mutable_input_data = *req->mutable_input_data();
-        auto& mutable_input_data_fields = *mutable_input_data.mutable_fields();
-
-        // TODO: Currently, this doesn't validate that we are passing the
-        // right input type in. We could query the metadata here and
-        // consult it.
-        for (const auto& kv : inputs) {
-            pb::Value& value = mutable_input_data_fields[kv.first];
-            mlmodel_details::tensor_to_pb_value(kv.second, &value);
```

Comment:
Please remove the declaration and implementation (and associated helper functions/types) of `mlmodel_details::tensor_to_pb_value`.